osxkeychain.go

```diff
@@ -346,12 +347,19 @@ func GetAllAccountNames(serviceName string) (accountNames []string, err error) {
 
 	defer C.CFRelease(resultsRef)
 
+	// Check that the resultsRef is an array
```

Comment:
Did you run into this, or is this just being cautious? If the latter, I'd add a comment saying something like using kSecMatchLimitAll is supposed to make the type of resultsRef an array, but double-check it.